tests/test_pip_repositories.py

```diff
@@ -132,6 +134,27 @@ def configure_auth(monkeypatch):
     monkeypatch.setenv("PIP_PASSWORD", _PRIVATE_REPO_PASSWORD)
 
 
+@pytest.fixture(autouse=True)
+def poetry_cache():
+    # This fixture serves to make sure that only one test that relies on the
+    # cache being cleared runs at a time
+
+    # We are going to rmtree the cache directory. Let's be extra careful to make
+    # sure we only delete a directory named "pypoetry-conda-lock" or one of its
+    # subdirectories.
+    to_delete = DEFAULT_CACHE_DIR.resolve()
+    assert to_delete.name == "pypoetry-conda-lock" or (
+        to_delete.parent.name == "pypoetry-conda-lock" and to_delete.name == "Cache"
+    )
+
+    # Grab a lock
+    with FileLock(to_delete.parent / ".conda_lock.lock"):
+        yield
+        # Do another independent check that triggers even if we're in optimized mode
+        if "pypoetry-conda-lock" in to_delete.parts:
```

Comment:
Oh, there's a risk of silent failure if this condition is false. We should fail hard.

What happens if the lockfile doesn't get cleaned up? Will the tests just silently hang waiting for release?

Reply:
Sounds good. I'll fail hard. Let me see if I can get something to clean it up better. I suspect they may hang but maybe what we can do is fail hard too (a timeout there) and I can maybe introduce a general cleanup at the end of the whole test.